### PR TITLE
Added Quirks for TS0601 _TZE200_fzo2pocs

### PR DESCRIPTION
**Pull request by blauret**
As requested in #653

### CODE REVIEW

**Comment by blauret** [comment]
@dmulcahey, Who can I ask to review merge this PR?

File: zhaquirks/tuya/ts0601.py [hunk 1]
```diff
@@ -55,3 +63,98 @@ class TuyaSingleSwitch(TuyaSwitch):
...
+    signature = {
+        # "node_descriptor": "<NodeDescriptor byte1=1 byte2=64 mac_capability_flags=142 manufacturer_code=4098
+        #                       maximum_buffer_size=82 maximum_incoming_transfer_size=82 server_mask=11264
+        #                       maximum_outgoing_transfer_size=82 descriptor_capability_field=0>",
+        # input_clusters=[0x0000, 0x0004, 0x0005, 0x000a, 0xef00]
+        # output_clusters=[0x0019]
+        # <SimpleDescriptor endpoint=1 profile=260 device_type=51 input_clusters=[0, 4, 5, 61184] output_clusters=[25]>
+        MODELS_INFO: [
+            ("_TZE200_fzo2pocs", "TS0601"),
+            ("_TZE200_xuzcvlku", "TS0601"),  # Not tested
```
Comment (rednus) [inline zhaquirks/tuya/ts0601.py:80]: Please move this to TuyaMoesCover0601 class.

Comment (blauret) [reply zhaquirks/tuya/ts0601.py]: Done

File: zhaquirks/tuya/__init__.py [hunk 2]
```diff
@@ -18,10 +19,24 @@
...
+#  0x2: open
+#  0x1: stop
+#  0x0: close
+# maps to
+# 0x0000: ("up_open", (), False),
+# 0x0001: ("down_close", (), False),
+# 0x0002: ("stop", (), False),
+
+TUYA_COVER_COMMAND = {0x0000: 0x0000, 0x0001: 0x0002, 0x0002: 0x0001}
+
```
Comment (rednus) [inline zhaquirks/tuya/__init__.py:39]: add below variable: 
`TUYA_SET_COVER_POSITION_COMMAND = 0x0002`

File: zhaquirks/tuya/__init__.py [hunk 3]
```diff
@@ -547,3 +562,116 @@ def handle_cluster_request(
...
+class TuyaWindowCoverControl(LocalDataCluster, WindowCovering):
+    """Manufacturer Specific Cluster of Device cover."""
+
+    def __init__(self, *args, **kwargs):
+        """Initialize instance."""
+        super().__init__(*args, **kwargs)
+        self.endpoint.device.cover_bus.add_listener(self)
+
+    def cover_event(self, command, value):
```
Comment (rednus) [inline zhaquirks/tuya/__init__.py:597]: update cover_event function to below: 
```
    def cover_event(self, command, value):
        """Cover event. update the position."""
        if value[0] == 4:
            position = 100 - value[4]
            self._update_attribute(ATTR_COVER_POSITION, position)
            self._update_attribute(self.attridx["current_position"], position)
```

File: zhaquirks/tuya/__init__.py [hunk 4]
```diff
@@ -547,3 +562,116 @@ def handle_cluster_request(
...
+        self,
+        hdr: foundation.ZCLHeader,
+        args: Tuple[TuyaManufCluster.Command],
+        *,
+        dst_addressing: Optional[
+            Union[t.Addressing.Group, t.Addressing.IEEE, t.Addressing.NWK]
+        ] = None,
+    ) -> None:
+        """Handle cluster request."""
+        tuya_payload = args[0]
```
Comment (rednus) [inline zhaquirks/tuya/__init__.py:580]: add debug line.. 
```
        _LOGGER.debug( "%s Received Attribute Report. Command is %x, Tuya Paylod values"
                       "[Status : %s, TSN: %s, Command: %s, Function: %s, Data: %s]", 
                       self.endpoint.device.ieee, hdr.command_id, tuya_payload.status, tuya_payload.tsn,
                       tuya_payload.command_id, tuya_payload.function, tuya_payload.data)
```

File: zhaquirks/tuya/__init__.py [hunk 5]
```diff
@@ -547,3 +562,116 @@ def handle_cluster_request(
...
+    def __init__(self, *args, **kwargs):
+        """Initialize instance."""
+        super().__init__(*args, **kwargs)
+        self.endpoint.device.cover_bus.add_listener(self)
+
+    def cover_event(self, command, value):
+        """Cover event. update the position."""
+        self._update_attribute(ATTR_COVER_POSITION, 100 - value[4])
+
+    def command(
```
Comment (rednus) [inline zhaquirks/tuya/__init__.py:601]: update command function to below: 
````
    def command(
        self,
        command_id: Union[foundation.Command, int, t.uint8_t],
        *args,
        manufacturer: Optional[Union[int, t.uint16_t]] = None,
        expect_reply: bool = True,
        tsn: Optional[Union[int, t.uint8_t]] = None,
    ):
        """Override the default Cluster command."""
        _LOGGER.debug( "%s Sending Tuya Cluster Command.. Cluster Command is %x, Arguments are %s", 
                       self.endpoint.device.ieee, command_id, args)
        tuya_payload = TuyaManufCluster.Command()
        # Open Close or Stop commands
        if command_id in (0x0000, 0x0001, 0x0002):
            tuya_payload.status = 0
            tuya_payload.tsn = 0
            tuya_payload.command_id = TUYA_CMD_BASE + self.endpoint.endpoint_id
            tuya_payload.function = 0
            tuya_payload.data = [
                1,
                TUYA_COVER_COMMAND[command_id],
            ]  # remap the command to the Tuya command          
        # Set Position Command
        elif command_id == 0x0005:
            tuya_payload = TuyaManufCluster.Command()
            tuya_payload.status = 0
            tuya_payload.tsn = 0
            tuya_payload.command_id = TUYA_SET_COVER_POSITION_COMMAND
            tuya_payload.function = 0
            position = args[0]
            tuya_payload.data = [
                4, 0, 0, 0, 
                100 - position,
            ]  
        # Custom Command
        elif command_id == 0x0006:
            tuya_payload = TuyaManufCluster.Command()
            tuya_payload.status = args[0]
            tuya_payload.tsn = args[1]
            tuya_payload.command_id = args[2]
            tuya_payload.function = args[3]
            tuya_payload.data = args[4]
        else: 
            tuya_payload = None
            
        if tuya_payload.command_id:

            _LOGGER.debug( "%s Sending Tuya Command. Paylod values [endpoint_id : %s, "
                           "Status : %s, TSN: %s, Command: %s, Function: %s, Data: %s]", 
                           self.endpoint.device.ieee, self.endpoint.endpoint_id, tuya_payload.status, 
                           tuya_payload.tsn, tuya_payload.command_id, tuya_payload.function, tuya_payload.data) 

            return self.endpoint.tuya_manufacturer.command(
                TUYA_SET_DATA, tuya_payload, expect_reply=True
            )
        else:
            _LOGGER.debug( "Unrecognised command: %x", command_id)
            return foundation.Status.UNSUP_CLUSTER_COMMAND
````

**Review by rednus — CHANGES_REQUESTED** [review]

**Comment by rednus** [comment]
As per the comments in #744  all working ok. 

@dmulcahey @Adminiuga could you please approve this merge?

**Pull request merged**